src/System.Net.HttpListener/src/System/Net/ListenerPrefix.cs

```diff
@@ -0,0 +1,176 @@
+//
+// System.Net.ListenerPrefix
+//
+// Author:
+//	Gonzalo Paniagua Javier (gonzalo@novell.com)
+//	Oleg Mihailik (mihailik gmail co_m)
+//
+// Copyright (c) 2005 Novell, Inc. (http://www.novell.com)
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+namespace System.Net
+{
+    internal sealed class ListenerPrefix
+    {
+        private string _original;
+        private string _host;
+        private ushort _port;
+        private string _path;
+        private bool _secure;
+        private IPAddress[] _addresses;
+        internal HttpListener _listener;
+
+        public ListenerPrefix(string prefix)
+        {
+            _original = prefix;
+            Parse(prefix);
+        }
+
+        public override string ToString()
+        {
+            return _original;
+        }
+
+        public IPAddress[] Addresses
+        {
+            get { return _addresses; }
+            set { _addresses = value; }
+        }
+        public bool Secure
+        {
+            get { return _secure; }
+        }
+
+        public string Host
+        {
+            get { return _host; }
+        }
+
+        public int Port
+        {
+            get { return _port; }
+        }
+
+        public string Path
+        {
+            get { return _path; }
+        }
+
+        // Equals and GetHashCode are required to detect duplicates in HttpListenerPrefixCollection.
+        public override bool Equals(object o)
+        {
+            ListenerPrefix other = o as ListenerPrefix;
+            if (other == null)
+                return false;
+
+            return (_original == other._original);
+        }
+
+        public override int GetHashCode()
+        {
+            return _original.GetHashCode();
+        }
+
+        private void Parse(string uri)
+        {
+            ushort default_port = 80;
+            if (uri.StartsWith("https://"))
```

Comment:
This is for basic parsing of the prefix, similar to windows parsing here: [AddPrefix](https://github.com/dotnet/corefx/blob/e61cda509c29326cdfb044d3e08c5b0ad388644d/src/System.Net.HttpListener/src/System/Net/HttpListener.Windows.cs#L346)